lib/gollum/assets.rb

```diff
@@ -44,3 +40,16 @@ def base_url
     end
   end
 end
+
+module Sprockets
```

Comment:
I don't think I understand the reasoning behind this monkeypatch. Can you explain why we need this?

Reply:
It's a bit complicated.

First of all, `//= require` is highly discouraged for modern sass because `//= require` embed the compiled child file directly in the parent. This has two problems:
1. It cause child files to be processed more than once
2. It completely breaks source map location
3. It slows down the I/O performance of sass backend

So I changed all `//= require` to `@use`.

The second problem is that `@use` cannot load `template.sass.erb`, because `@use` is directly processed by sass backend, which does not know how to process `erb`. From what I can see the only use case for `erb` is this `rocticon_css` function. Therefore, I replaced the `erb` with a custom sprocket `sassc` function, which is now directly invoked by sass backend.

Reply:
In other words, this is the preferred way of calling ruby function from sass, instead of using `erb`.

It effectively replaces this erb helper (the helper is also used by js code, so it's kept there): https://github.com/gollum/gollum/blob/476c7de094732115c553337eff487750724eeae9/lib/gollum/views/helpers.rb#L78-L83

Reply:
Reference: https://github.com/rails/sprockets/blob/v4.2.1/lib/sprockets/sassc_processor.rb#L89-L100

Reply:
thanks for the explanation @ntkme! That makes sense.

We'd like to revisit our workflow for building gollum's SASS a little before merging this. At the moment we are asking people to add the precompiled and minimized CSS/JS to their PRs, but that is obviously suboptional for a few reasons. We would prefer to have a workflow for where only changes to the SASS/JS templates are directly contributed, and a build step is triggered to precompile from the new source.

Let's see if we get around to doing this this week. Or if you feel like it and have time, you're welcome to contribute a Github Actions file that achieves this, as well. :)

Reply:
@dometto Added a commit to automate that in github actions. You can see that it creates a commit authored by `github-actions[bot]` in this PR. Also updated some outdated actions that are broken.

Reply:
Thanks a lot for the contribution, looks great! Small question in the review.